File: leetcode/2236_root_equals_sum_of_children/2236_root_equals_sum_of_children.js

// Definition for a binary tree node.
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {boolean}
 */
 var checkTree = function(root) {
    if ((root.left.val + root.right.val) === root.val) return true
    return false  
};

const rootList = [5,3,1];
let treeNode = new TreeNode(val = rootList[0]);
treeNode.left = new TreeNode(val = rootList[1]);
treeNode.right = new TreeNode(val = rootList[2]);

console.log(checkTree(treeNode));

File: leetcode/2236_root_equals_sum_of_children/2236_root_equals_sum_of_children.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def checkTree(self, root: Optional[TreeNode]) -> bool:
        if ((root.left.val + root.right.val) == root.val):
            return True
        return False


if __name__ == "__main__":
    root_list = [5, 3, 1]
    tree_node = TreeNode(val=root_list[0])
    tree_node.left = TreeNode(val=root_list[1])
    tree_node.right = TreeNode(val=root_list[1])

    print(Solution().checkTree(tree_node))
